app/bundles/LeadBundle/Entity/LeadRepository.php

```diff
@@ -578,6 +578,34 @@ public function getEntities($args = array())
             ) : $results;
     }
 
+
+    public function getEntitiesLight($args = array()){
+    	$limit = 20000; //lines
+    	$results = array();
+    	$offset = 0;
+    	$result = $this->requestEntitiesLight($args, $offset, $limit);
+    	$results = array_merge($results, $result);
+    	while (count($result) === $limit){
+    		$offset += $limit;
+    		$result = $this->requestEntitiesLight($args, $offset+1, $limit);
+    		$results = array_merge($results, $result);
+    	}
+    	return $results;
+    }
```

Comment:
@escopecz I didn't write this code. I'm really discovering it as you are. I think the idea was to stream packets of 20000 contacts at a time. But as I can see, this effectively does not alleviate the problem.
I think the only optimization you'll find here is this narrower selection which consumes less memory.

I think we are not taking full advantage of Doctrine. And we really should be. I'll look further into that tomorrow.